book/custom_commands.md

```diff
@@ -162,6 +204,38 @@ Now, we can call this updated definition using the shorthand flag:
 > greet -a 10 hello
 ```
 
+## Rest parameters
+
+There may be cases when you want to define a command which takes any number of positional arguments. We can do this with a rest parameter, using the following `...` syntax:
+
+```nushell
+def greet [...name: string] {
+  echo "hello all:"
+  for $n in $name {
+    echo $n
+  }
+}
+
+greet earth mars jupiter venus
+```
+
+We could call the above definition of the `greet` command with any number of arguments, including none at all. All of the arguments are collected into `$name` as a list.
+
+Rest parameters can be used together with positional parameters:
+
+```
+def greet [vip: string, ...name: string] {
+  echo "hello to our VIP" $vip
+  echo "and hello to everybody else:"
+  for $n in $name {
+    echo $n
+  }
+}
+
+#     $vip          $name
+#     .--. .----------------------.
```

Comment:
Are these lines supposed to be here?

Reply:
Ah, yep - those are supposed to highlight which arguments get passed into which parameters. (The first line covers only the first argument and is marked `$vip`, while the second line covers the rest of them which are passed into `$name`.) I can see how it might look a tad strange! Is there a clearer way this could be shown?

Reply:
Ah OK, I get it now - wasn't quite sure what was going on at first glance. I think some of that might be that I didn't realize the `.` and `-` are just parts of the same line; this looks a tiny bit better to me but let me know which way you prefer and I can merge the PR either way.

```suggestion
#     $vip          $name
#     ---- ------------------------
```

Reply:
I prefer that too, thanks!